tutoraspects/templates/aspects/apps/superset/pythonpath/performance_metrics.py

```diff
@@ -78,16 +87,33 @@ def performance_metrics(course_key, print_sql):
 
         logger.info("Waiting for clickhouse log...")
         time.sleep(20)
-        get_query_log_from_clickhouse(report, print_sql)
+        get_query_log_from_clickhouse(report, query_contexts, print_sql)
         return report
 
 
-def measure_chart(slice, extra_filters=[]):
-    """
-    Measure the performance of a chart and return the results.
-    """
-    logger.info(f"Fetching slice data: {slice}")
-    query_context = json.loads(slice.query_context)
+def get_query_contexts_from_assets():
+    query_contexts = {}
+
+    for root, dirs, files in os.walk(CHART_PATH):
+        for file in files:
+            if not file.endswith(".yaml"):
+                continue
+
+            path = os.path.join(root, file)
+            with open(path, "r") as file:
+                asset = yaml.safe_load(file)
+                if "query_context" in asset and asset["query_context"]:
+                    query_contexts[asset["uuid"]] = json.loads(asset["query_context"])
+
+    logger.info(f"Found {len(query_contexts)} query contexts")
+    return query_contexts
```

Comment:
what about returning a dictionary with all the context matched against the UUID. Then use it to access it directly below